app/controllers/teachers_controller.rb

```diff
@@ -18,23 +18,27 @@ def new
   def edit; end
 
   def create
-    @new_user = User.new(user_params.merge(password: new_user_temp_password))
-    # TODO: Add support to create students user
-
-    if @new_user.save
-      redirect_to teacher_path(@user),
-                  notice: "#{@new_user.role.capitalize} was created with password #{new_user_temp_password}."
-    else
-      render :new, status: :unprocessable_entity
+    ActiveRecord::Base.transaction do
+      @new_user = User.create(user_params.merge(password: new_user_temp_password))
+      if @new_user.student?
+        StudentInformation.create(student_information_params.merge(user: @user))
+        GroupMember.create(user: @new_user, group_id: params[:user][:group])
+      end
     end
+
+    return redirect_to_teacher if @new_user.save
+
+    render :new, status: :unprocessable_entity
   end
 
   def update
     if @user.update(user_params)
-      redirect_to teacher_url(@user), notice: "#{@user.role.capitalize} was successfully updated."
-    else
-      render :edit, status: :unprocessable_entity
+      redirect_url = teacher_url(@user)
+      redirect_url = teacher_students_url(current_user.id) if @user.student?
```

Comment:
same here, in which scenario would we update a student here? I think something doesn't sound right

Reply:
Fixed!